sensors/utils/utils.go

```diff
@@ -0,0 +1,42 @@
+// Package utils contains helper functions for the sensor implementations.
+package utils
+
+import (
+	"context"
+	"image"
+
+	"github.com/edaniels/gostream"
+	"github.com/pkg/errors"
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/rimage"
+	"go.viam.com/rdk/utils"
+)
+
+// GetPNGImage first attempts to get a lazy PNG image. If the image is not a lazy PNG, the
+// function expects it to be an YCBCR image. If it is neither a lazy PNG nor an YCBCR, the
+// function errors out.
+func GetPNGImage(ctx context.Context, cam camera.Camera) ([]byte, func(), error) {
+	// We will hint that we want a PNG.
+	// The Camera service server implementation in RDK respects this; others may not.
+	readImgCtx := gostream.WithMIMETypeHint(ctx, utils.WithLazyMIMEType(utils.MimeTypePNG))
+	img, release, err := camera.ReadImage(readImgCtx, cam)
+	if err != nil {
+		return nil, nil, err
+	}
+	if lazyImg, ok := img.(*rimage.LazyEncodedImage); ok {
+		if lazyImg.MIMEType() != utils.MimeTypePNG {
+			return nil, nil, errors.Errorf("expected mime type %v, got %T", utils.MimeTypePNG, img)
```

Comment:
Why is this not `return nil, nil, errors.Errorf("expected mime type %v, got %T", utils.MimeTypePNG, lazyImg.MIMEType())`?

Reply:
Would that still be `%T`? I think MimeType() returns a string, so I think you would change the `%T` to `%v`. Not confident though.

Reply:
I don't think we need to replace `img` with `lazyImg.MIMEType()`, since `img` will not be of mime type `png` if we get to this line.